app/views/browse/pod.html

```diff
@@ -124,19 +124,60 @@
                     ng-init="containers = pod.status.containerStatuses"
                     ng-if="containersRunning(pod.status.containerStatuses) > 0">
                     <uib-tab-heading>Terminal</uib-tab-heading>
-                    <div>
+
+                    <div class="mar-bottom-md">
                       <span class="pficon pficon-info" aria-hidden="true"></span>
-                      When you navigate away from this pod, any processes running
-                      in these terminals will quit.
+                      Terminal names with <i class="fa fa-terminal btn-success"></i> are active, healthy terminal sessions to living containers.
+                      They will remain connected until you navigate away from this page.
                     </div>
-                    <div class="pod-container-terminal"
-                      ng-repeat="container in containers | orderBy:'name' track by container.name"
-                      ng-if="container.state.running">
-                      <h3 ng-if="containers.length > 1">{{container.name}}</h3>
-                      <kubernetes-container-terminal pod="pod" container="container.name"
-                        prevent="!terminalTabWasSelected">
-                      </kubernetes-container-terminal>
+
+                    <div>
+                      <ui-select
+                        ng-model="selected"
+                        on-select="onTerminalSelectChange(selected)"
+                        class="mar-left-none pad-left-none col-xs-3">
+                        <ui-select-match
+                          placeholder="Select or search">
```

Comment:
I would probably just say "Container name" then it explains what it is you are looking at.   "Select or search" isn't a very useful placeholder, plus its not really search, its typeahead.